packages/flow-core/src/components/f-suggest/f-suggest.ts

```diff
@@ -223,6 +255,116 @@ export class FSuggest extends FRoot {
+				break;
+			case "Enter":
+				event.preventDefault();
+				this.selectOption();
+				break;
+			case "Escape":
+				event.preventDefault();
+				this.popOverElement.open = false;
+				break;
+		}
+	}
+
+	navigateOptions(direction: number) {
+		if (this.isSuggestionArray) {
+			const totalOptions = this.filteredSuggestions?.length;
+			if (totalOptions === 0) return;
+
+			// Calculate the next index based on the direction
+			const newIndex = this.currentIndex + direction;
+
+			// Ensure the new index stays within bounds
+			this.currentIndex =
+				((newIndex as number) + (totalOptions as number)) % (totalOptions as number);
+
+			// Optionally, you can scroll the dropdown to bring the selected option into view if it's outside the viewport.
+			this.scrollFocusedOptionIntoView();
+		} else {
+			if (this.filteredSuggestions) {
+				const totalCategories = Object.keys(this.filteredSuggestions).length;
+				if (totalCategories === 0) return;
+
+				const currentCategory = Object.keys(this.filteredSuggestions)[this.currentCategoryIndex];
+				const totalOptions = (this.filteredSuggestions as FSuggestSuggestionsCategory)[
+					currentCategory
+				].length;
+
+				// Calculate the next option index based on the direction
+				const newIndex = this.currentIndex + direction;
+
+				// Handle navigation within the current category
+				if (newIndex >= 0 && newIndex < totalOptions) {
+					this.currentIndex = newIndex;
+				} else if (newIndex >= totalOptions) {
+					// Move to the next category
+					this.currentCategoryIndex = (this.currentCategoryIndex + 1) % totalCategories;
+					this.currentIndex = 0; // Set the first option of the new category as focused
+				} else {
+					// Move to the previous category
+					this.currentCategoryIndex =
+						(this.currentCategoryIndex - 1 + totalCategories) % totalCategories;
+					this.currentIndex =
+						(this.filteredSuggestions as FSuggestSuggestionsCategory)[currentCategory].length - 1; // Set the last option of the new category as focused
+				}
+
+				// Optionally, you can scroll the dropdown to bring the selected option into view if it's outside the viewport.
+				this.scrollFocusedOptionIntoView();
+			}
+		}
+	}
+
+	scrollFocusedOptionIntoView() {
+		const optionElements = this.shadowRoot?.querySelectorAll(".f-select-options-clickable");
+		if (optionElements) {
+			if (optionElements.length > this.currentIndex) {
+				optionElements[this.currentIndex].scrollIntoView({
+					behavior: "auto", // 'auto' or 'smooth' for scrolling behavior
+					block: "nearest" // Scroll to the nearest edge of the container
+				});
+			}
+		}
+	}
+
+	selectOption() {
+		if (this.isSuggestionArray) {
+			if (this.filteredSuggestions) {
+				if (this.currentIndex >= 0 && this.currentIndex < this.filteredSuggestions.length) {
+					const selectedOption = (this.filteredSuggestions as string[] | FSuggestTemplate[])[
+						this.currentIndex
+					];
+					if (this.isSuggestionObjects) {
```

Comment:
This if condition will never execute since you have `(this.isSuggestionArray`on line 340